torch/distributions/binomial.py

```diff
@@ -80,15 +77,22 @@ def probs(self):
     def param_shape(self):
         return self._param.size()
 
+    def _get_homogeneous_count(self):
+        total_count = int(self.total_count.max().item())
+        if not self.total_count.min() == total_count:
+            raise NotImplementedError("Inhomogeneous total count not supported by method.")
+        return total_count
+
     def sample(self, sample_shape=torch.Size()):
-        shape = self._extended_shape(sample_shape) + (self.total_count,)
+        total_count = self._get_homogeneous_count()
```

Comment:
Gosh I guess since we're already summing `Bernoulli`s we could draw inhomogeneous samples via
```py
def sample(self, sample_shape=torch.Size()):
    with torch.no_grad():
        max_count = self.total_count.max().item()
        shape = self._extended_shape(sample_shape) + (max_count,)
        bernoullis = torch.bernoulli(self.probs.unsqueeze(-1).expand(shape))
        if self.total_count.min() != max_count:
            arange = torch.arange(max_count, out=self.total_count.new_empty(max_count))
            bernoullis *= (arange < self.total_count.unsqueeze(-1)).type_as(bernoullis)
        return bernoullis.sum(dim=-1)
```
WDYT?

Reply:
I was thinking about it. :) It may lead to us to creating some large intermediate tensors, but it should work fine! Let me make that change and then we can get in @apaszke's opinion.